src/tests/simulation/tests/integration_tests/recovery/background_ddl.rs

```diff
@@ -86,14 +118,78 @@ async fn test_background_mv_barrier_recovery() -> Result<()> {
 
     sleep(Duration::from_secs(10)).await;
 
-    // Make sure after finished, we should have 5000_000 rows.
     session
         .run("SELECT COUNT(v1) FROM m1")
         .await?
-        .assert_result_eq("600000");
+        .assert_result_eq("700000");
+
+    // Make sure that if MV killed and restarted
+    // it will not be dropped.
 
     session.run("DROP MATERIALIZED VIEW m1").await?;
     session.run("DROP TABLE t1").await?;
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_background_ddl_cancel() -> Result<()> {
+    use std::env;
+    env::set_var(
+        "RUST_LOG",
+        "info,risingwave_meta=debug,risingwave_stream::executor::backfill=debug",
+    );
+    env::set_var("BACKFILL_SNAPSHOT_READ_DELAY", "100");
+    tracing_subscriber::fmt()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .with_ansi(false)
+        .init();
```

Comment:
The setting of env var and global logger will be applied on the process level. I'm not sure whether there's side effect as tests might run in a multi-threaded manner.

Reply:
BTW, I believe `madsim` has helped us to initialize the the logger. (cc @wangrunji0408 correct me if I'm wrong) And theoretically the second `init` call will panic. We may use scoped tracing dispatcher instead.

https://github.com/risingwavelabs/risingwave/blob/e392db0c6f468caf89597294a94d4e4134c61eaf/src/connector/benches/nexmark_integration.rs#L131

Reply:
> I'm not sure whether there's side effect as tests might run in a multi-threaded manner.

True. I will mitigate it by setting it just before creating the mv, and unsetting it immediately after.

Reply:
Seems like it will cause test to fail. Given that most of the integration test do not ingest a lot of data, I think it's fine for now. I will set the delay to 50ms.

After runji updates madsim I will try rate limit approach again.

Reply:
> I believe `madsim` has helped us to initialize the the logger

Ah, seems not true anymore after https://github.com/madsim-rs/madsim/issues/96.

Reply:
(If main-cron tle, we can always separate background_ddl tests in ci).